src/main/java/dan200/computercraft/shared/peripheral/modem/wired/WiredModemElement.java

```diff
@@ -61,4 +62,31 @@ public void networkChanged( @Nonnull IWiredNetworkChange change )
     protected abstract void attachPeripheral( String name, IPeripheral peripheral );
 
     protected abstract void detachPeripheral( String name );
+
+    public boolean isSharingPeripherals()
+    {
+        return sharingPeripherals;
+    }
+
+    public boolean setSharingPeripherals( boolean enabled, boolean updateTile )
+    {
+        if( sharingPeripherals == enabled ) return sharingPeripherals;
+        sharingPeripherals = enabled;
+        if( updateTile )
```

Comment:
Doing this as a boolean flag feels super janky. It's probably worth having two methods, where one of them is a `trySetSharingPeripherals` or something.

Reply:
Done. Now it's two methods.